web_external/Datasets/UploadDatasetMetadataView.js

```diff
@@ -0,0 +1,195 @@
+            // Folder already created
+            this.startUpload(this.uploadFolder);
+        } else {
+            // Create new upload folder with unique name
+            this.uploadFolder = new girder.models.FolderModel({
+                name: 'isic_upload_' + Date.now(),
+                parentType: 'user',
+                parentId: girder.currentUser.id,
+                description: 'ISIC uploads'
+            });
+
+            this.uploadFolder.once('g:saved', function () {
+                this.startUpload(this.uploadFolder);
+            }, this).once('g:error', function () {
+                girder.events.trigger('g:alert', {
+                    icon: 'cancel',
+                    text: 'Could not create upload folder.',
+                    type: 'error',
+                    timeout: 4000
+                });
+            }, this).save();
+        }
+    },
+
+    uploadFinished: function (files) {
+        this.uploadedCsvFiles = _.pluck(files.files, 'name');
+        this.updateUploadWidget();
+    },
+
+    startUpload: function (folder) {
+        // Configure upload widget and begin upload
+        this.uploadWidget.parentType = 'folder';
+        this.uploadWidget.parent = folder;
+        this.uploadWidget.uploadNextFile();
+    },
+
+    updateUploadWidget: function () {
+        var visible = false;
+        var uploadList = [];
+        if (this.uploadedCsvFiles.length) {
+            visible = false;
+            uploadList = this.uploadedCsvFiles;
+        } else {
+            visible = true;
+        }
+
+        this.$('.isic-upload-widget-container').toggle(visible);
+        this.$('.isic-upload-reset-container').toggle(!visible);
+
+        this.uploadWidget.render();
+        this.$('.isic-upload-list').text(
+            'Uploaded: ' + uploadList.join(', '));
+    },
+
+    resetUpload: function () {
+        // Delete uploaded files
+        this.uploadFolder.once('g:success', function () {
+            this.uploadedCsvFiles = [];
+            this.updateUploadWidget();
+        }, this).removeContents();
+    },
+
+    registerMetadata: function (metadataFileId) {
+        this.dataset.registerMetadata(metadataFileId);
+    }
+});
+
+isic.router.route('uploadDatasetMetadata/:id', 'uploadDatasetMetadata', function (id) {
+    if (girder.currentUser) {
+        // Registered users must:
+        //  (1) Accept the TOS
+        //  (2) Request and receive create dataset access
+        // before being able to see the upload dataset metadata view
+        if (!isic.models.UserModel.currentUserCanAcceptTerms()) {
+            girder.events.trigger('g:navigateTo', isic.views.TermsAcceptanceView);
+        } else if (!girder.currentUser.canCreateDataset()) {
+            girder.events.trigger('g:navigateTo', isic.views.UploadDatasetRequestView);
+        } else {
+            // Fetch the dataset, then navigate to the view
+            var dataset = new isic.models.DatasetModel();
+            dataset.set({
```

Comment:
I agree it's nicer to do that; I updated the code. The only issue is that effectively makes `dataset` an unused variable, since it's used only in the chained function calls, but this passes the style check. I believe omitting the variable assignment would necessitate manually disabling a style check for that line, which is uglier IMO.